src/wavpack_local.h

```diff
@@ -97,121 +99,8 @@ typedef struct {
     unsigned char *ape_tag_data;
 } M_Tag;
 
-// RIFF / wav header formats (these occur at the beginning of both wav files
-// and pre-4.0 WavPack files that are not in the "raw" mode)
-
-typedef struct {
-    char ckID [4];
-    uint32_t ckSize;
-    char formType [4];
-} RiffChunkHeader;
-
-typedef struct {
-    char ckID [4];
-    uint32_t ckSize;
-} ChunkHeader;
-
-#define ChunkHeaderFormat "4L"
-
-typedef struct {
-    uint16_t FormatTag, NumChannels;
-    uint32_t SampleRate, BytesPerSecond;
-    uint16_t BlockAlign, BitsPerSample;
-    uint16_t cbSize, ValidBitsPerSample;
-    int32_t ChannelMask;
-    uint16_t SubFormat;
-    char GUID [14];
-} WaveHeader;
-
-#define WaveHeaderFormat "SSLLSSSSLS"
-
-////////////////////////////// WavPack Header /////////////////////////////////
-
-// Note that this is the ONLY structure that is written to (or read from)
-// WavPack 4.0 files, and is the preamble to every block in both the .wv
-// and .wvc files.
-
-typedef struct {
-    char ckID [4];
-    uint32_t ckSize;
-    int16_t version;
-    unsigned char block_index_u8;
-    unsigned char total_samples_u8;
-    uint32_t total_samples, block_index, block_samples, flags, crc;
-} WavpackHeader;
-
-#define WavpackHeaderFormat "4LS2LLLLL"
```

Comment:
Just passing by with curiosity - why the removal of this macro and the many below? I wouldn't call that an optimisation - they'll get stripped away by the compiler.

Reply:
I agree with and like this change and was going to merge it soon (I've been overwhelmed with day job recently).

However I just read your comment about the major ABI bump, which is kind of a show-stopper for me. Are you sure about that? The WavpackContext pointer gets changed from a void pointer to an opaque pointer, but in either case the user of libwavpack treats it exactly the same (i.e. they can only check it for NULL and pass it back to the library, no dereferencing of any kind). I'm sure that binaries would continue to work unmodified, but I'm sure that from some viewpoint the type did change even if they're equivalent. What do you think?

I'm sure it would also be possible to make an equivalent change without modifying wavpack.h at all, but it would be a little ugly and I like the additional type checking that this provides users of libwavpack.

Reply:
Or is it an issue that the binaries _would_ actually break because of some sort of name-mangling and/or linker type-checking going on under the hood that I am blissfully ignorant of? Guess I could try it...at least on Linux.